src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +34,173 @@ ActorInit En_Encount3_InitVars = {
+}
+
+void func_809AD058(EnEncount3* this) {
+    this->childActorId = ACTOR_EN_JSO;
+    this->unk150 = 1;
+    this->unk15E = 0x1E;
+    this->actionFunc = func_809AD084;
+}
+
+void func_809AD084(EnEncount3* this, PlayState* play) {
+    if (this->switchFlag >= 0 && Flags_GetSwitch(play, this->switchFlag)) {
+        Actor_Kill(&this->actor);
+    } else if (!(this->unk16C < this->actor.xzDistToPlayer) && (Player_GetMask(play) == PLAYER_MASK_GARO) &&
+               !D_809AD810) {
+        if (this->unk15E > 0) {
+            this->unk15E--;
+        } else {
+            this->child =
+                Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, this->childActorId, this->actor.world.pos.x,
+                                   this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, this->childParams);
+            if (this->child != NULL) {
+                this->unk14E++;
+                D_809AD810 = true;
+                this->actionFunc = func_809AD194;
+            }
+        }
+    }
+}
+
+void func_809AD194(EnEncount3* this, PlayState* play) {
+    if (this->unk14E == 0) {
+        this->unk178 = 0.0f;
+        if (this->switchFlag >= 0) {
+            Flags_SetSwitch(play, this->switchFlag);
+        }
+        this->actionFunc = func_809AD1EC;
+    }
+}
+
+void func_809AD1EC(EnEncount3* this, PlayState* play) {
+    if (this->unk174 < 0.002f) {
+        D_809AD810 = false;
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnEncount3_Update(Actor* thisx, PlayState* play2) {
+    EnEncount3* this = THIS;
+    f32 new_var;
+    PlayState* play = play2;
+    Player* player = GET_PLAYER(play);
+    f32 sp3C;
+    f32 sp38;
+
+    this->actionFunc(this, play);
+    if (this->actionFunc == func_809AD194) {
+        new_var = this->unk16C * 0.52f;
+        if (new_var < this->actor.xzDistToPlayer) {
+            this->unk178 = ((this->unk16C * 0.03125f) * 0.001f) + 0.1f;
+        } else {
+            this->unk178 = 0.06f;
+        }
+
+        if (((this->unk16C + 50.0f) + BREG(0)) < this->actor.xzDistToPlayer) {
+            if (this->actionFunc == func_809AD194) {
+                this->unk148 = 0;
+                this->unk178 = 0.0f;
+                D_809AD810 = false;
+                if (((this->child != NULL) && (this->child->update != NULL)) && (this->child->colChkInfo.health > 0)) {
+                    Actor_Kill(this->child);
+                    this->child = NULL;
+                }
+                func_809AD058(this);
+            }
+        } else if (this->unk16C < this->actor.xzDistToPlayer) {
+            s16 i;
+
+            for (i = 0; i < PLAYER_BODYPART_MAX; i++) {
+                player->flameTimers[i] = Rand_S16Offset(0, 0xC8);
+            }
+            player->isBurning = true;
+
+            sp3C = this->actor.world.pos.x - player->actor.world.pos.x;
+            sp38 = this->actor.world.pos.z - player->actor.world.pos.z;
+            if (Play_InCsMode(play) == 0) {
```

Comment:
```suggestion
            if (!Play_InCsMode(play)) {
```
Bool can be like this